torch/_inductor/compile_fx.py

```diff
@@ -1576,6 +1588,21 @@ def fx_codegen_and_compile(
         )
         scheme = _AsyncFxCompile(scheme)
 
+    if fx_compile_progressive:
```

Comment:
It seems like async is a subtype of progressive (it's just a progressive where -O0 is eager, right?) So maybe async should be killed in favor of control flags to progressive...